src/react-components/info-dialog.js

```diff
@@ -27,6 +27,21 @@ class InfoDialog extends Component {
 
     const loc = document.location;
     this.shareLink = `${loc.protocol}//${loc.host}${loc.pathname}`;
+    this.onKeyPress = this.onKeyPress.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener("keypress", this.onKeyPress);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("keypress", this.onKeyPress);
+  }
+
+  onKeyPress(e) {
+    if (e.key === "Escape") {
```

Comment:
can you check `document.activeElement.matches('body, a')`? (or you can check `e.target` too. there are a couple ways to test this. the `a` is because the original footer link still has focus when the `click` event has fired to open the modal but the user has not interacted with the page/modal. usually, `document.activeElement` would be `document.body`.)

in the _Get Updates_ modal, when a user has focused the `<input>` field for the email address, pressing the `Escape` key should not close the dialog.

![image](https://user-images.githubusercontent.com/203725/39729608-add1be06-5211-11e8-8e56-54c85b649108.png)

Reply:
Hmm, I am not sure about that one. I would still expect the modal to close even if I was in the textbox if I hit excape. Checking for the "a" doesnt seem right because I could click somewhere inside the dialog and then no longer have an "a" focused.

Reply:
for accessibility reasons, we try to not hijack keys when form fields are focused. in past projects, we do something [like this](https://github.com/mozilla/marketplace-core-modules/blob/59bb600/core/utils.js#L63-L66).

if it's not too much trouble, can you add a tiny helper function to handle this?

```js
var formTags = /input|keygen|meter|option|output|progress|select|textarea/i;
function fieldFocused (e) {
  return formTags.test(e.target.nodeName);
}
```

and then use as such:

```js
if (!fieldFocused(e.target) && e.key === "Escape") {
```